fuse_core/include/fuse_core/util.h

```diff
@@ -340,6 +395,264 @@ static inline void quaternionToAngleAxis(const double * q, double * angle_axis,
   }
 }
 
+/**
+ * @brief Compute the jacobian of a quaternion normalization
+ *
+ * @param[in] q The quaternion
+ * @return      The jacobian of the quaternion normalization
+ */
+static inline fuse_core::Matrix4d jacobianQuatNormalization(Eigen::Quaterniond q)
+{
+  fuse_core::Matrix4d ret;
+  ret <<
+     q.x() * q.x() + q.y() * q.y() + q.z() * q.z(), -q.w() * q.x(), -q.w() * q.y(), -q.w() * q.z(),
+     -q.x() * q.w(), q.w() * q.w() + q.y() * q.y() + q.z() * q.z(), -q.x() * q.y(), -q.x() * q.z(),
+     -q.y() * q.w(), -q.y() * q.x(), q.w() * q.w() + q.x() * q.x() + q.z() * q.z(), -q.y() * q.z(),
+     -q.z() * q.w(), -q.z() * q.x(), -q.z() * q.y(), q.w() * q.w() + q.x() * q.x() + q.y() * q.y();
+
+  ret /= std::pow(q.norm(), 3.);
+  return ret;
+}
+
+/**
+ * @brief Convert a pose covariance in x, y, z, roll, pitch, yaw to a pose covariance in x, y, z, qw, qx, qy, qz
+ *
+ * @param[in] pose    The pose which covariance should be converted
+ * @param[in] cov_rpy The covariance in x, y, z, roll, pitch, yaw
+ * @return            The pose covariance in x, y, z, qw, qx, qy, qz
+ */
+static inline fuse_core::Matrix7d convertToPoseQuatCovariance(
```

Comment:
Well the name is missleading it seams it calculates the jacobian for the pose

Reply:
No, it does not just compute the jacobian for the pose but then uses the jacobian to convert the covariance represented in [x, y, z, roll, pitch, yaw] to a covariance in [x, y, z, qw, qx, qy, qz]. The conversion itself happens in the last line of the function.

Reply:
ah ok